### PR TITLE
Draft : [ONEToolchain] Introduce ONEToolchain

### PR DESCRIPTION
**Pull request by Psalmist-KIM**
Introduce ONEToolchain Draft code.

Related Issue : #1271 #1272 

ONE-vscode-DCO-1.0-Signed-off-by: Nakhyun Kim [dss02094@gmail.com](mailto:dss02094@gmail.com)

### CODE REVIEW

File: src/Backend/ONE/ONEToolchain.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    this.installedToolchains = this.getToolchains(toolchainType).filter((toolchain) => {
+      return parseInt(execSync(`dpkg-query --show ${toolchain.info.name} > /dev/null 2>&1; echo $?`)
+                          .toString()
+                          .trim()) === 0;
+    });
+    return this.installedToolchains;
+  }
+
+  prerequisitesForGetToolchains(): Command {
+    return new Command('/bin/echo', ['prerequisites']);
```
[the remark this answers is not in the document]
Comment (YeonJeongLee00) [reply src/Backend/ONE/ONEToolchain.ts:75]: We're going to implement this function after the Debian package is completed.
so should we comment on it?

[the remark this answers is not in the document]
Comment (jyoungyun) [reply src/Backend/ONE/ONEToolchain.ts:73]: Then you don't need to implement this function now. 

```suggestion
    throw Error('Invalid prerequisitesForGetToolchains call');
```

**Comment by jyoungyun** [comment]
You can merge this PR even though docker and debian package is not ready.

How about `error-thrown` in the `getToolchains`, `getInstalledToolchains` and `prerequisitesForGetToolchains` functions and the first merge the PR defined only by `ONEToolchain`? Including the tests modified by registering `ONEToolchain`. :)

**Comment by lettersto** [comment]
> How about `error-thrown` in the `getToolchains`, `getInstalledToolchains` and `prerequisitesForGetToolchains` functions and the first merge the PR defined only by `ONEToolchain`? Including the tests modified by registering `ONEToolchain`. :)

We're going to merge the PR soon!

**Pull request closed without merging**